gatsby-node.js

```diff
@@ -90,7 +108,41 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
     const { name } = node;
     createPage({
       path: name,
-      component: path.resolve('src', 'templates', 'CopyPage.js'),
+      component: CopyPage,
+      context: { id },
+    });
+  });
+
+  // Blog data
+  const posts = result.data.blog.nodes;
+  const postsPerPage = 10;
+  const pageCount = Math.ceil(posts.length / postsPerPage);
+
+  // Blog index pages
+  Array.from({ length: pageCount }).forEach((_, i) => {
+    const currentPage = i + 1;
+    createPage({
+      path: i === 0 ? '/blog/' : `/blog/${currentPage}`,
+      component: BlogIndex,
+      context: {
+        limit: postsPerPage,
+        skip: i * postsPerPage,
+        pageCount,
+        currentPage,
+      },
+    });
+  });
+
+  // Blog posts
+  posts.forEach((node) => {
+    const { id } = node.childMarkdownRemark;
+    const { name, relativeDirectory } = node;
+    // Build blog post path from a combination of directory and name.
+    // If the filename is `index.md`, just use directory.
+    const url = path.posix.join(relativeDirectory, name.replace('index', ''));
+    createPage({
+      path: `/blog/${url}`,
+      component: CopyPage,
       context: { id },
     });
   });
```

Comment:
Is there a way to just create a Gatsby page for every Markdown file in a directory automatically? I thought `gatsby-source-filesystem` might do this without any manual setup.

Reply:
Nope!